### DISCUSSION TITLE
Add new permission to handle Environment Admins

### DISCUSSION THREAD

**Discussion by Danielecina** [mia-platform/community#31] [Ideas]
It would enable us to have per-person, temporary, testing environments

**Comment by fredmaggiowski** [comment]
Hi, what do you mean with "handle environment admins"? It's already possible to only allow specific users to change environment by making actions such as deploy/restart pod/manage dashboards etc; what is the feature you'd like to have?

**Reply by Danielecina** [reply]
Yes, it is sure. I've been too hermetic in my proposal.
In the future, if we enabled temporary environments, we could ensure that all users, from dev to maintainer, can create environments from the new environments section. But, once completed, they must also be able to cancel them. 
But only the admin can do it to delete the environment.
How to fix this flux? Then, the idea of the environment administrators

**Reply by fredmaggiowski** [reply]
A specific permission for managing temporary environments could be created and provided to more roles (such as developers too) but strong constraints must be enforced to make sure such environments are actually temporary. A user then would be able to create a temporary environment and delete only its own temporary environments.

**Reply by Danielecina** [reply]
But on the other hand, it is a bit too permissible to allow a user to be able to create all the time. then it can be useful to have temporary permissions or a temporary role in a slot time where it allows him to do this above, no?

**Reply by fredmaggiowski** [reply]
for sure, but these are two different features

**Reply by fredmaggiowski** [reply]
Anyhow I'd suggest you to write these two features in the internal product roadmap so that they can be discussed and planned :)
